### DISCUSSION TITLE
add version info to `preferred_technologies.md`

### DISCUSSION THREAD

**Discussion by wusixuan0** [Matt-Dionis/nlad#5] [General]
Hi! I came here from the Reddit post. really comprehensive guide!

I just thought under the `preferred_technologies.md` in Required Documentation, you could add version information.
and during initial planning, let LLM decide if additional context (like API docs) is needed based on version info.

**Comment by Matt-Dionis** [comment]
I like this idea a lot. It would become even more helpful if the LLM being called had access to a web search tool. That way, it could find issues with specific versions of a dependency.

**Reply by wusixuan0** [reply]
To let LLM search internet and fix issues with specific versions of a dependency, maybe using fetch MCP[https://github.com/modelcontextprotocol/servers/tree/main/src/fetch] for Claude desktop might work? For general bug fixes, I saw people use either Claude desktop MCP or AI agents to automatically pull issues and fix them. I couldn't find the exact post I saw. Windsurf, [a Sonnet-powered IDE by Codeium](https://www.anthropic.com/customers/codeium) , might be able to accomplish that as well (I haven’t explore Windsurf but they collaborate with Claude).

In my workflow, I would dump the issue-related content I saw online (GitHub issues, Stack Overflow, release notes) to LLM and fix it from there.
